src/ldp/agent/react_agent.py

```diff
@@ -111,12 +117,49 @@ class ReActAgent(BaseModel, Agent[SimpleAgentState]):
             " results in fewer action selection failures."
         ),
     )
+    planning: bool = Field(
+        default=False,
+        description=(
+            "Specifies whether to use planning mode. When enabled, the agent will"
+            " first output a step-by-step plan as a checklist, then think about the"
+            " next immediate action to take. The plan is updated after each step."
+        ),
+    )
+    planning_thought: bool = Field(
+        default=False,
+        description=(
+            "Specifies whether to use thought generation in planning mode. When enabled,"
+            " the agent will generate critic/plan/thought, and when disabled, the agent only generate"
+            " critic/plan. Currently not recommended due to causing tool use to be overly verbose."
+        ),
+    )
+
+    # New fields for ReActPlanningModule prompts
+    critic_prompt: str = Field(
+        default=REACT_PLANNING_CRITIC_PROMPT,
+        description="Prompt for critic assessment in planning mode.",
+    )
+
+    plan_prompt: str = Field(
```

Comment:
Can you write some unit tests of this, and use VCR cassettes to cache the HTTP requests